Feat/migration guide (#240)
nmshd/documentation: _docs_integrate/version-5.md

---
# Start automatic generation
permalink: integrate/version-5

> published: false

Reviewer: should be true.

title: "Version 5"
type: scenario
toc: true
properties:
  - id: SC115
  - category: Migration Guides

>   - description: Änderungen bzgl. Version 5

Reviewer: german description?

Author: I change this, however in the UseCaseV5 there was something in german, when I filtered so I was confused.

Author: https://jssoftgmbh.sharepoint.com/:x:/r/sites/Enmeshed/_layouts/15/Doc.aspx?sourcedoc=%7B485D26E0-DA69-45D3-A25E-922CD9CA8290%7D&file=UseCasesV5.xlsx&wdOrigin=TEAMS-MAGLEV.p2p_ns.rwc&action=default&mobileredirect=true

Reviewer: I see.

This change enables the Sender of the Request to specify more precisely who the `owner` of a RelationshipAttribute should be when querying it and better reflects real application scenarios.

> ## Changed and new Use Cases

Reviewer: ```suggestion
## Changed and added Use Cases
```

### RevokeRelationship

> With Version 5 the Use Case RevokeRelationship has been added. So now it is possible to revoke a RelationshipRequest which has already the `status` `"pending"`.

Reviewer: ```suggestion
With Version 5 the Use Case RevokeRelationship has been added. So now it is possible to revoke a Relationship which is in status `"Pending"` and was created by yourself.
```

should we also explain how this works? Or route to the usecase page?

This change enables the Sender of the Request to specify more precisely who the `owner` of a RelationshipAttribute should be when querying it and better reflects real application scenarios.

## Changed and added Use Cases

### RevokeRelationship

> With Version 5 the Use Case RevokeRelationship has been added. So now it is possible to revoke a RelationshipRequest which is in `status` `"pending"` and was created by yourself.

Reviewer: ```suggestion
With Version 5 the Use Case RevokeRelationship has been added. So now it is possible to revoke a RelationshipRequest which is in status `"Pending"` and was created by yourself.
```

You also ignored my other part. I miss more information how to actually Revoke a relationship.

Author: I didn't ignored it I wanted to respond you that I can add the use case tomorrow and link on it if thats okay for you?

Reviewer: sure, that's good. So we merge this now and you add the details in another PR?

- RevokeRelationshipReactivation: Revokes the reactivation of the terminated [Relationship]({% link _docs_integrate/data-model-overview.md %}#relationship) with the given id.
- TerminateRelationship: Terminates the active Relationship [Relationship]({% link _docs_integrate/data-model-overview.md %}#relationship) with the given id.

### Renaming of the Use Case GetSharedVersionsOfRepositoryAttribute to GetSharedVersionsOfAttribute

- Taking ThirdPartyRelationshipAttributes into account, we wanted to extend the functionality of this use case for [RelationshipAttributes]({% link _docs_integrate/data-model-overview.md %}#relationshipattribute).
- The use case GetSharedVersionsOfAttribute was already added in v4 and GetSharedVersionsOfRepositoryAttribute was marked as deprecated. With v5 it is deleted.

## New Events

There are several Events which has been added. For example regarding the RelationshipTermination the following has been added:

- `"transport.relationshipDecomposedBySelf"`
- `"transport.relationshipReactivationCompleted"`

> - `"transport.relationshipReactivationRequested"`

Reviewer: I think you forgot to add these events to _docs_integrate/connector-events.md

Reviewer: and the event OutgoingRequestFromRelationshipCreationChangeCreatedAndCompletedEvent is now called OutgoingRequestFromRelationshipCreationCreatedAndCompletedEvent

Reviewer: `consumption.outgoingRequestFromRelationshipCreationChangeCreatedAndCompleted` -> `consumption.outgoingRequestFromRelationshipCreationCreatedAndCompleted`

Author: I have added them but I think it would be necessary that in the newer future someone should have a look over the events which are listed there and update all.

Reviewer: Yep, the combination is fine :)
